libbeat/processors/add_process_metadata/add_process_metadata.go

```diff
@@ -271,6 +271,7 @@ func (p *processMetadata) toMap() common.MapStr {
 			"pid":        p.pid,
 			"ppid":       p.ppid,
 			"start_time": p.startTime,
+			"username":   p.username,
```

Comment:
I suggest not to set this field when p.username is the empty string.

Reply:
done